api/repositories/k8s_client.go

```diff
@@ -57,7 +57,7 @@ func (f UnprivilegedClientFactory) BuildClient(authInfo authorization.Info) (cli
 	userClient, err := client.NewWithWatch(config, client.Options{})
 	if err != nil {
 		if k8serrors.IsUnauthorized(err) {
-			return nil, authorization.InvalidAuthError{}
+			return nil, authorization.NotAuthenticatedError{}
```

Comment:
Just noting from a convo we had during our sync that the Kubernetes errors are named for the HTTP status codes `401 Unauthorized` and `403 Forbidden` which we use to represent "not authenticated" and "not authorized" respectively. Interesting! 🤓 

* IsUnauthorized: https://github.com/kubernetes/apimachinery/blob/276a8a7530a30f42c82b1f0fa2261a905130bf97/pkg/api/errors/errors.go#L664
* IsForbidden: https://github.com/kubernetes/apimachinery/blob/276a8a7530a30f42c82b1f0fa2261a905130bf97/pkg/api/errors/errors.go#L678

Reply:
In the light of mimicking the CF API as closely as possible I think I favour this change. Here is why:
- As @tcdowney pointed out in this [thread](https://cloudfoundry.slack.com/archives/C0297673ASK/p1644946622354189) the Cloud Controller would only ever return a `403` error if the token is valid, has not yet expired and the user described by this token has insufficient permissions to perform the action.
- The `BuildClient` method is usually being called from a repo, meaning that the auth middleware was already able to parse the token (i.e. it is valid), but k8s rejected it with a `401`. Having a valid token rejected by k8s can mean two things - the token has expired, the token's issuer is not trusted by the cluster. The middleware only guarantees that the token is valid and well formed, not that it is issued by a trusted issuer. I think this case is similar to the token having expired or not having been issued by the correct UAA in which case the right error to return should be `401` or `NotAuthenticatedError`

Reply:
In this case I think changing the error type to a `NotAuthenticatedError` is justified